jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ContextProvider.java

```diff
@@ -131,161 +136,186 @@ public ContextProvider()
     public boolean isExtractWars()
     {
-        return isExtract();
-    }
-
-    /**
-     * @return True if WAR and JAR are extraced on deploy.
-     */
-    @ManagedAttribute("extract WAR and JAR files")
-    public boolean isExtract()
-    {
-        return _extract;
+        return Boolean.parseBoolean(_properties.get("jetty.deploy.extractWars"));
     }
 
     /**
      * Set the extractWars.
+     * This is equivalent to the <code>jetty.deploy.extractWars</code> property.
      *
-     * @param extract the extractWars to set
-     * @deprecated use {@link #setExtract(boolean)}
+     * @param extractWars the extractWars to set
      */
-    @Deprecated
-    public void setExtractWars(boolean extract)
+    public void setExtractWars(boolean extractWars)
     {
-        setExtract(extract);
-    }
-
-    /**
-     * Set to extract WAR and JAR files.
-     *
-     * @param extract the extractWars to set
-     */
-    public void setExtract(boolean extract)
-    {
-        _extract = extract;
+        _properties.put("jetty.deploy.extractWars", Boolean.toString(extractWars));
     }
 
     /**
      * Get the parentLoaderPriority.
+     * This is equivalent to the <code>jetty.deploy.parentLoaderPriority</code> property.
      *
      * @return the parentLoaderPriority
      */
     @ManagedAttribute("parent classloader has priority")
     public boolean isParentLoaderPriority()
     {
-        return _parentLoaderPriority;
+        return Boolean.parseBoolean(_properties.get("jetty.deploy.parentLoaderPriority"));
     }
 
     /**
      * Set the parentLoaderPriority.
+     * This is equivalent to the <code>jetty.deploy.parentLoaderPriority</code> property.
      *
      * @param parentLoaderPriority the parentLoaderPriority to set
      */
     public void setParentLoaderPriority(boolean parentLoaderPriority)
     {
-        _parentLoaderPriority = parentLoaderPriority;
+        _properties.put("jetty.deploy.parentLoaderPriority", Boolean.toString(parentLoaderPriority));
     }
 
     /**
      * Get the defaultsDescriptor.
+     * This is equivalent to the <code>jetty.deploy.defaultsDescriptor</code> property.
      *
      * @return the defaultsDescriptor
      */
     @ManagedAttribute("default descriptor for webapps")
     public String getDefaultsDescriptor()
     {
-        return _defaultsDescriptor;
+        return _properties.get("jetty.deploy.defaultsDescriptor");
     }
 
     /**
      * Set the defaultsDescriptor.
+     * This is equivalent to the <code>jetty.deploy.defaultsDescriptor</code> property.
      *
      * @param defaultsDescriptor the defaultsDescriptor to set
      */
     public void setDefaultsDescriptor(String defaultsDescriptor)
     {
-        _defaultsDescriptor = defaultsDescriptor;
-    }
-
-    public ConfigurationManager getConfigurationManager()
-    {
-        return _configurationManager;
+        _properties.put("jetty.deploy.defaultsDescriptor", defaultsDescriptor);
```

Comment:
If you're going to be referencing the same property a couple of times in this file, then make a constant for it. In general referencing property names from xml files is a little fragile, but I understand why it's being done. I guess ide refactoring will make any future changes to those property names less problematic.